QuasarHelios2015/src/org/team1540/quasarhelios/ControlInterface.java

```diff
@@ -35,23 +39,65 @@ public static void setupJoysticks() {
         DriveCode.octocanumShiftingButton = Igneous.joystick1.getButtonSource(1);
         DriveCode.recalibrateButton = Igneous.joystick1.getButtonSource(2);
 
-        Igneous.joystick2.getButtonSource(1).send(Elevator.setTop);
-        Igneous.joystick2.getButtonSource(2).send(Elevator.setBottom);
-        Igneous.joystick2.getButtonSource(3).send(Elevator.stop);
-        BooleanMixing.pumpWhen(QuasarHelios.globalControl, Igneous.joystick2.getButtonChannel(4), Elevator.overrideEnabled);
-        Igneous.joystick2.getAxisSource(1).send(Elevator.overrideValue);
+        Igneous.joystick2.getButtonSource(4).send(Elevator.setTop);
+        Igneous.joystick2.getButtonSource(1).send(Elevator.setBottom);
+        Igneous.joystick2.getButtonSource(2).send(Elevator.stop);
+        BooleanMixing.pumpWhen(QuasarHelios.globalControl, Igneous.joystick2.getButtonChannel(5), Elevator.overrideEnabled);
 
-        Igneous.joystick2.getButtonSource(5).send(Rollers.runRollersButton);
-        Igneous.joystick2.getButtonSource(6).send(Rollers.toggleRollersButton);
-        Igneous.joystick2.getButtonSource(7).send(Rollers.toggleOpenButton);
-//        Igneous.joystick2.getAxisSource(5).send(Rollers.controlArmsIndependently);
-//        Igneous.joystick2.getAxisSource(6).send(Rollers.controlRollersManually);
+        FloatInput axis2 = Igneous.joystick2.getAxisSource(2);
+        axis2.send(Elevator.overrideValue);
+        
+        EventInput povPressed = BooleanMixing.onPress(Igneous.joystick2.isPOVPressedSource(1));
+        FloatInput povAngle = Igneous.joystick2.getPOVAngleSource(1);
+        EventInput up = EventMixing.filterEvent(FloatMixing.floatIsInRange(povAngle, -0.1f, 0.1f), true, povPressed);
+        EventInput down = EventMixing.filterEvent(FloatMixing.floatIsInRange(povAngle, 179.9f, 180.1f), true, povPressed);
+        EventInput left = EventMixing.filterEvent(FloatMixing.floatIsInRange(povAngle, 269.9f, 270.1f), true, povPressed);
+        EventInput right = EventMixing.filterEvent(FloatMixing.floatIsInRange(povAngle, 89.9f, 90.1f), true, povPressed);
+        
+        up.send(() -> {
+            if (Rollers.direction.get()) {
+                Rollers.running.set(true);
+            } else {
+                Rollers.running.set(!Rollers.running.get());
+            }
+            
+            Rollers.direction.set(false);
+        });
+        
+        down.send(() -> {
+            if (!Rollers.direction.get()) {
+                Rollers.running.set(true);
+            } else {
+                Rollers.running.set(!Rollers.running.get());
+            }
+            
+            Rollers.direction.set(true);
+        });
```

Comment:
You may be able to do some of these through things like `Rollers.direction.setFalseWhen`.